packages/sprotty-protocol/src/actions.ts

```diff
@@ -0,0 +1,500 @@
+            bounds: options.bounds,
+            requestId: generateRequestId()
+        };
+    }
+}
+
+/**
+ * Sent from the model source to the client to display a popup in response to a RequestPopupModelAction.
+ * This action can also be used to remove any existing popup by choosing EMPTY_ROOT as root element.
+ */
+export interface SetPopupModelAction extends ResponseAction {
+    kind: typeof SetPopupModelAction.KIND
+    newRoot: SModelRoot
+}
+export namespace SetPopupModelAction {
+    export const KIND = 'setPopupModel';
+
+    export function create(newRoot: SModelRoot, requestId?: string): SetPopupModelAction {
+        return {
+            kind: KIND,
+            newRoot,
+            responseId: requestId!
+        };
+    }
+}
+
+/**
+ * Sent from the model source (e.g. a DiagramServer) to the client to update the bounds of some
+ * (or all) model elements.
+ */
+export interface SetBoundsAction extends Action {
+    kind: typeof SetBoundsAction.KIND
+    bounds: ElementAndBounds[]
+}
+export namespace SetBoundsAction {
+    export const KIND = 'setBounds';
+
+    export function create(bounds: ElementAndBounds[]): SetBoundsAction {
+        return {
+            kind: KIND,
+            bounds
+        };
+    }
+}
+
+/**
+ * Sent from the model source to the client to request bounds for the given model. The model is
+ * rendered invisibly so the bounds can derived from the DOM. The response is a ComputedBoundsAction.
+ * This hidden rendering round-trip is necessary if the client is responsible for parts of the layout
+ * (see `needsClientLayout` viewer option).
+ */
+export interface RequestBoundsAction extends RequestAction<ComputedBoundsAction> {
+    kind: typeof RequestBoundsAction.KIND
+    newRoot: SModelRoot
+}
+export namespace RequestBoundsAction {
+    export const KIND = 'requestBounds';
+
+    export function create(newRoot: SModelRoot): RequestBoundsAction {
+        return {
+            kind: KIND,
+            newRoot,
+            requestId: generateRequestId()
+        };
+    }
+}
+
+/**
+ * Sent from the client to the model source (e.g. a DiagramServer) to transmit the result of bounds
+ * computation as a response to a RequestBoundsAction. If the server is responsible for parts of
+ * the layout (see `needsServerLayout` viewer option), it can do so after applying the computed bounds
+ * received with this action. Otherwise there is no need to send the computed bounds to the server,
+ * so they can be processed locally by the client.
+ */
+export interface ComputedBoundsAction extends ResponseAction {
+    kind: typeof ComputedBoundsAction.KIND
+    bounds: ElementAndBounds[]
+    revision?: number
+    alignments?: ElementAndAlignment[]
+}
+export namespace ComputedBoundsAction {
+    export const KIND = 'computedBounds';
+
+    export function create(bounds: ElementAndBounds[], options: { revision?: number, alignments?: ElementAndAlignment[], requestId: string }): ComputedBoundsAction {
```

Comment:
RequestId should be an optional option:
```suggestion
    export function create(bounds: ElementAndBounds[], options: { revision?: number, alignments?: ElementAndAlignment[], requestId?: string }): ComputedBoundsAction {
```